_posts/2020-06-30-Making-IrishGen-SPARQL-Part-II.md

```diff
@@ -0,0 +1,180 @@
+cannot finish in a reasonable time or finish at all in the face of the
+query timeout restrictions.  In Stardog, this can be overcome by
+turning reasoning off when doing an exploration of the graph if the
+query does not return in a reasonable time but this limits what is
+possible.  It is the user's discretion to understand what their query
+involves and what compromises they are willing to tolerate.
+
+With reasoning turned off, one can explore the graph a bit at a time
+as one can see below:
+
+<img src="{{site.baseurl}}/assets/images/construct_baeth_2.png" />
+
+A much more ambitious graph to construct is the graph of Baeth's
+ancestors or to reconstruct his genealogical line from the
+Triplestore.  This query introduces a few new constructs which are
+handy when working with SPARQL.
+
+```sparql
+prefix rel: <http://purl.org/vocab/relationship/>
+
+construct {
+    ?s rel:childOf ?parent.
+    ?y rel:parentOf ?baeth.
+} 
+from <tag:stardog:api:context:all>
+where {
+    values ?baeth {<http://example.com/Rawl_B502/clann_aingeda.trig#Báeth> }
+
+    {
+      ?s rel:ancestorOf ?baeth;
+         rel:childOf ?parent
+    } union {
+      ?y rel:parentOf ?baeth
+    }
+}
+```
+
+The `constuct` form first shows what will be the product of the query.
+The `values` keyword is used to store static URLs which will be used
+repeatedly within the query.  This cuts down on visual complexity
+overall of the query.  The next part of the query is the meat of the
+process.  This informally says: "return all people who are the
+ancestors of Baeth and their parent-child relationship and return the
+parent of Baeth as well".  These are then reconstructed in the
+`construct` form to create the reconstructed genealogy of Baeth.  The
+`union` is used to get Baeth themselves attached to the output through
+their parent which makes it slightly easier to interpret the
+visualisation.
+
+<img src="{{site.baseurl}}/assets/images/construct_baeth_3.png" />
+
+One will note that the genealogy only goes back to Imchada.  This is
+due to the fact that Imchada is not linked to any other part of the
+genealogies.  The reason for this is outside the scope of this post
+but could be as simple as a missing `owl:sameAs` link from elsewhere.
+The curators tend to be conservative and if there is no clear evidence
+that two individuals are the same, the policy is to leave them as is.
+Finally, a note about performance.  This query took 21000 milliseconds
+or about 21 seconds to run.  Reconstructing this by hand would take
+much longer and creating a visualisation even longer still.  Resulting
+in an interactive visualisation is just one more additional benefit.
+
+As we can see, `construct` queries allow the transformation of RDF
+graphs found in the Triplestore into other RDF graphs.  Using this, we
+can extract interesting sub-graphs from the dataset and visualize them
+using Stardog Studio.  This allows the reconstruction of genealogical
```

Comment:
And other packages, presumably? #ProductPlacement

Reply:
If this really is a unique feature of Stardog Studio, perhaps say so, as we have previously given the impression that there are multiple Triplestores to choose from.